pkg/flowkit/config/json/network.go

```diff
@@ -28,60 +29,102 @@ type jsonNetworks map[string]jsonNetwork
 
 // transformToConfig transforms json structures to config structure.
 func (j jsonNetworks) transformToConfig() (config.Networks, error) {
-	var networks config.Networks
+	networks := make(config.Networks, 0)
 
 	for networkName, n := range j {
-		network := config.Network{
-			Name: networkName,
-			Host: n.Host,
+		if n.Advanced.NetworkKey != "" && n.Advanced.Host != "" {
+			networks = append(networks, config.Network{
+				Name:       networkName,
+				Host:       n.Advanced.Host,
+				NetworkKey: n.Advanced.NetworkKey,
+			})
+		} else if n.Advanced.Host != "" {
+			networks = append(networks, config.Network{
+				Name: networkName,
+				Host: n.Advanced.Host,
+			})
+		} else if n.Simple.Host != "" {
+			networks = append(networks, config.Network{
+				Name: networkName,
+				Host: n.Simple.Host,
+			})
+		} else {
+			return nil, fmt.Errorf("failed to transform config missing network host")
 		}
-
-		networks = append(networks, network)
 	}
 
 	return networks, nil
 }
 
-// transformToJSON transforms config structure to json structures for saving.
+// transformNetworksToJSON transforms config structure to json structures for saving.
 func transformNetworksToJSON(networks config.Networks) jsonNetworks {
 	jsonNetworks := jsonNetworks{}
 
 	for _, n := range networks {
-		jsonNetworks[n.Name] = jsonNetwork{
-			Host: n.Host,
+		if n.NetworkKey != "" {
+			jsonNetworks[n.Name] = transformAdvancedNetworkToJSON(n)
+		} else {
+			jsonNetworks[n.Name] = transformSimpleNetworkToJSON(n)
 		}
 	}
 
 	return jsonNetworks
 }
 
+func transformSimpleNetworkToJSON(n config.Network) jsonNetwork {
+	return jsonNetwork{
+		Simple: simpleNetwork{
+			Host: n.Host,
+		},
+	}
+}
+
+func transformAdvancedNetworkToJSON(n config.Network) jsonNetwork {
+	return jsonNetwork{
+		Advanced: advancedNetwork{
+			Host:       n.Host,
+			NetworkKey: n.NetworkKey,
+		},
+	}
+}
+
 type jsonNetwork struct {
-	Host string
+	Simple   simpleNetwork
+	Advanced advancedNetwork
+}
+
+type simpleNetwork struct {
+	Host string `json:"host"`
 }
 
 type advancedNetwork struct {
-	Host  string
-	Chain string
+	Host       string `json:"host"`
+	NetworkKey string `json:"key"`
```

Comment:
In some location I'm seeing `host,key` but in some locations I'm seeing `host,network-key`. Is this ok?

Reply:
https://github.com/onflow/flow-cli/pull/449/commits/05b28f3a8666def59064eebb64f01d7102dc1e6c